onnxruntime/core/providers/tvm/tvm_runner_impl.cc

```diff
@@ -0,0 +1,180 @@
+  dst_inds.reserve(num);
+  for (auto& info : inputs_info_) {
+    // TODO(vvchernov): decomposition declaration only available with -std=c++1z or -std=gnu++1z
+    auto& i = info.first;
+    auto& shape = info.second;
+    const OrtValue* input_tensor = ort.KernelContext_GetInput(context, i);
+    ORT_ENFORCE(input_tensor->IsTensor());
+    const Tensor& tensor = input_tensor->Get<Tensor>();
+    const OrtDevice& device = tensor.Location().device;
+    auto tensor_info = ort.GetTensorTypeAndShape(input_tensor);
+    auto tensor_type = ort.GetTensorElementType(tensor_info);
+    ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+
+    DLTensor t;
+    t.device = GetDLDevice(device);
+    t.dtype = GetDataType(tensor_type);
+    t.strides = nullptr;
+    t.byte_offset = 0;
+    t.data = const_cast<void*>(ort.GetTensorData<void>(input_tensor));
+    t.ndim = shape.size();
+    t.shape = shape.data();
+    dst.emplace_back(t);
+    dst_inds.push_back(i);
+  }
+}
+
+void RunnerImpl::add_device_type_data2output_tensors(Ort::CustomOpApi& ort,
+                                                     OrtKernelContext* context) {
+  size_t num_outputs = output_tensors_.size();
+  for (auto i = 0u; i < num_outputs; i++) {
+    //setup output tensor property
+    OrtValue* output_tensor = ort.KernelContext_GetOutput(context,
+                                                          i,
+                                                          output_shapes_[i].data(),
+                                                          output_shapes_[i].size());
+    ORT_ENFORCE(output_tensor->IsTensor());
+    const Tensor& tensor = output_tensor->Get<Tensor>();
+    const OrtDevice& device = tensor.Location().device;
+    auto tensor_info = ort.GetTensorTypeAndShape(output_tensor);
+    auto tensor_type = ort.GetTensorElementType(tensor_info);
+    ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+
+    output_tensors_[i].device = GetDLDevice(device);
+    output_tensors_[i].dtype = GetDataType(tensor_type);
+    output_tensors_[i].data = ort.GetTensorMutableData<void>(output_tensor);
+  }
+}
+
+bool RunnerImpl::compare_shapes(const TVMTensorShape& shape1, const TVMTensorShape& shape2) const {
+  size_t size = shape1.size();
+  if (shape2.size() == size) {
+    for (size_t i = 0; i < size; ++i) {
+      if(shape1[i] != shape2[i]) {
+        return false;
+      }
+    }
+  } else {
+    return false;
+  }
+
+  return true;
+}
+
+/* ------------------------------------ GERunnerImpl ------------------------------------ */
+
+GERunnerImpl::GERunnerImpl(const std::shared_ptr<TvmModule>& mod,
+                           const InputsInfoMap& inputs_info,
+                           const TVMTensorShapes output_shapes,
+                           const std::vector<DLTensor> output_tensors) :
+  RunnerImpl(mod, inputs_info, output_shapes, output_tensors) {
+}
+
+void GERunnerImpl::set_input(Ort::CustomOpApi& ort, OrtKernelContext* context) {
+  std::vector<size_t> inds;
```

Comment:
use InlinedVector

Reply:
I can not use InlinedVector for indeces due to their size is number of input tensors. But it is unknown value. Of course for the majority of neural networks it is one, but there are multiple input architectures which can have any amount of inputs.